File: yospace/src/main/java/com/bitmovin/player/integration/yospace/YospaceAdStartedEvent.java
package com.bitmovin.player.integration.yospace;

import com.bitmovin.player.api.event.data.AdStartedEvent;
import com.bitmovin.player.config.advertising.AdSourceType;
import com.bitmovin.player.model.advertising.Ad;

public class YospaceAdStartedEvent extends AdStartedEvent {
    private boolean isTruex;

    public YospaceAdStartedEvent(AdSourceType clientType, String clickThroughUrl, int indexInQueue, double duration, double timeOffset, String position, double skipOffset, boolean isTruex, Ad ad) {
        super(clientType, clickThroughUrl, indexInQueue, duration, timeOffset, position, skipOffset, ad);
        this.isTruex = isTruex;
    }

    public boolean isTruex() {
        return isTruex;
    }
}

File: yospace/src/main/java/com/bitmovin/player/integration/yospace/YospaceAdStartedEvent.kt
package com.bitmovin.player.integration.yospace

import com.bitmovin.player.api.advertising.AdSourceType

class YospaceAdStartedEvent(
    clientType: AdSourceType = AdSourceType.Unknown,
    clickThroughUrl: String,
    indexInQueue: Int,
    duration: Double,
    timeOffset: Double = 0.0,
    position: String = "position",
    skipOffset: Double = 0.0,
    ad: Ad? = null,
    val companionAds: List<CompanionAd> = emptyList()
): CustomEvent()
